### DISCUSSION TITLE
Why stub pages for /src/pages?

### DISCUSSION THREAD

**Discussion by mattkrick** [agustinusnathaniel/nextarter-chakra#102] [Q&A] [answered]
I see this template uses stub pages in src/pages that just point to src/components/pages.
I'm sure there's a good reason for it, I'd love to learn the backstory!
My guess is that it with all the components living in src/components it makes relative imports shorter & running CLI commands on all components easier. Is there another reason?

**Comment by agustinusnathaniel** [comment, marked as answer]
Hi @mattkrick 👋 

Your guess is one of the reasons, another reason is `/pages` folder in next.js have its own rules (for how to defined the page, embedding GSSP, GSSR, etc). 

If someday I'd like to copy pages components into another project which maybe not using next.js (for example vite with react-router or remix), I'd just need to copy `lib` folder and re-create the `/pages` file conforming with the other project's rules.

Portability and to separate which folder is next.js specific (src/pages) and which are not (src/lib)

I'm open for suggestions though. I've been tinkering with various foldering conventions and finding sweet spot in `src/lib` and `src/pages` foldering structure (highly influenced by svelte kit template foldering with `src/lib` and `src/routes` where we can put stuffs in `src/lib` and `src/routes` is svelte kit specific pages folder).
